### PR TITLE
adds up/down sampling blocks

### PR DESCRIPTION
**Pull request by wyli**
### Description
some necessary building blocks for #396 

### Status
**Ready**

### Types of changes

- [x] non-Breaking change new features
- [x] New tests added to cover the changes
- [x] Docstrings/Documentation updated

### CODE REVIEW

File: tests/test_upsample_block.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+                "scale_factor": s,
+                "align_corners": True,
+            },
+            torch.randn(16, 3, 4, 5, 6),
+        ]
+        test_case.append(expected_shape)
+        TEST_CASES_EQ.append(test_case)
+
+
+class TestMaxAvgPool(unittest.TestCase):
```
Comment (Nic-Ma) [inline tests/test_upsample_block.py:61]: I think it should be "TestUpsample"

File: monai/networks/blocks/upsample.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+            with_conv: whether to use a transposed convolution for upsampling. Defaults to False.
+            mode: the interpolation mode. Defaults to "linear", this corresponds to linear, bilinear, trilinear
+                for 1D, 2D, and 3D respectively.
+            align_corners: set the align_corners parameter of `torch.nn.Upsample`. Defaults to True.
+        """
+        super().__init__()
+        if not out_channels:
+            out_channels = in_channels
+        if not with_conv:
+            _mode = ["linear", "bilinear", "trilinear"][spatial_dims - 1] if mode.lower().endswith("linear") else mode
```
Comment (Nic-Ma) [inline monai/networks/blocks/upsample.py:49]: May I know why you do this mode selection logic instead of just using the `mode` parameter?
Thanks.

Comment (wyli) [reply monai/networks/blocks/upsample.py:49]: Yes, not all the modes are compatible with all data shapes: https://github.com/pytorch/pytorch/blob/71af538e31547e5b1bc814c9e00323a21905baf3/torch/nn/functional.py#L3155-L3170

Comment (ericspod) [reply monai/networks/blocks/upsample.py:49]: This makes sense but I would put `["linear", "bilinear", "trilinear"]` into a separate local variable `linear_modes` or something like that to improve readability.

**Pull request merged**